compiler/codegen/library.cpp

```diff
@@ -483,15 +508,85 @@ static void makePYFile() {
     fprintf(py.fptr, "\t\tExtension(\"%s\",\n", pythonModulename);
     fprintf(py.fptr, "\t\t\tinclude_dirs=[numpy.get_include()],\n");
     fprintf(py.fptr, "\t\t\tsources=[\"%s.pyx\"],\n", pythonModulename);
-    fprintf(py.fptr, "\t\t\tlibraries=[\"%s\"] + chpl_libraries)))\n",
-            libname.c_str());
+    if (startsWithLib) {
+      // Cython includes anything listed in the libraries variable in the
+      // compileline with a -l prefix.  The -l prefix only works if the library
+      // is named `libname.*`; `name.a` will not be found for `-lname`, so
+      // only include the generated library file here if its name starts with
+      // lib
+      fprintf(py.fptr, "\t\t\tlibraries=[\"%s\"] + chpl_libraries)))\n",
+              libname.c_str());
+    } else {
+      fprintf(py.fptr, "\t\t\tlibraries=chpl_libraries)))\n");
+    }
 
     gGenInfo->cfile = save_cfile;
   }
   // Don't "beautify", it will remove the tabs
   closeLibraryHelperFile(&py, false);
 }
 
+// Once all the python files have been generated and the .a/.so has been made,
+// make the Python module!
+void codegen_make_python_module() {
+  std::string getOldPythonPath = "echo $PYTHONPATH";
+  std::string pythonPath = runCommand(getOldPythonPath);
+  if (pythonPath == "\n") {
+    pythonPath = CHPL_RUNTIME_INCL;
+  } else {
+    pythonPath.erase(pythonPath.length() - 1);
+    pythonPath += ":";
+    pythonPath += CHPL_RUNTIME_INCL;
+  }
+  pythonPath += "/python";
+
+  std::string getCFlags = "$CHPL_HOME/util/config/compileline --cflags";
+  std::string cFlags = runCommand(getCFlags);
```

Comment:
Should 'getCompilelineOption' be used instead?

Reply:
getCompilelineOption also replaces long paths in the string with $(CHPL_HOME), etc, which aren't valid in a bash command (but are in the generated Makefile, where getCompilelineOption is primarily used).  I could split the portion prior to those cleanups into a separate function, but I was getting tired of rearranging functions so I didn't at the time.

Reply:
OK 👍